bee-node/bee-plugin/bee-plugin-version-checker/src/lib.rs

```diff
@@ -5,43 +5,167 @@
     type Error = Infallible;
 
-    async fn start(node: &mut N, _config: Self::Config) -> Result<Self, Self::Error> {
+    async fn start(node: &mut N, current_version: Self::Config) -> Result<Self, Self::Error> {
         node.spawn::<Self, _, _>(|shutdown| async move {
-            info!("Running.");
+            log::info!("Running.");
 
             let mut ticker = ShutdownStream::new(
                 shutdown,
                 IntervalStream::new(interval(Duration::from_secs(CHECK_INTERVAL_SEC))),
             );
 
             while ticker.next().await.is_some() {
-                // TODO
+                if let Some(release_info) = get_release_info().await {
+                    show_version_status(&current_version, release_info).await;
+                }
             }
 
-            info!("Stopped.");
+            log::info!("Stopped.");
         });
 
         Ok(Self::default())
     }
 }
+
+/// Alerts the user (through a `WARN` level log event) if there is a newer release version of Bee available,
+/// and provides a URL to the release's summary webpage.
+async fn show_version_status(current_version: &str, release_info: Vec<ReleaseInfoBuilder>) {
+    let current_version = match semver::Version::parse(current_version) {
+        Err(e) => {
+            log::error!("Error parsing current Bee version ({}): {}.", current_version, e);
+            return;
+        }
+        Ok(current_version) => current_version,
+    };
+
+    match latest(release_info, !current_version.pre.is_empty()) {
+        Some(latest) => {
+            if latest.version > current_version {
+                log::warn!(
+                    "Found a more recent release version ({}), available at {}.",
+                    latest.version,
+                    latest.html_url,
+                );
+            } else {
+                log::info!("On the latest release version ({}).", current_version);
+            }
+        }
+        None => log::warn!("Could not identify the latest release version."),
+    }
+}
+
+/// Returns a list of [`ReleaseInfoBuilder`]s describing all Github releases if the API request and
+/// deserialization was successful. Otherwise, returns `None`.
+async fn get_release_info() -> Option<Vec<ReleaseInfoBuilder>> {
+    let client = reqwest::Client::builder().user_agent("bee").build().unwrap();
+
+    match client
+        .get("https://api.github.com/repos/iotaledger/bee/releases")
+        .send()
+        .await
+        .and_then(|resp| resp.error_for_status())
+    {
+        Ok(resp) => match resp.json::<Vec<ReleaseInfoBuilder>>().await {
+            Ok(releases) => Some(releases),
+            Err(e) => {
+                // Error deserializing response.
+                log::error!("Error deserializing Github API reponse: {}", e);
+                None
+            }
+        },
+        Err(e) => {
+            // Error occured during API request.
+            log::error!("Error performing Github API request: {}", e);
```

Comment:
Should we maybe just `warn` the user if a request fails? It could happen that GitHub is down, which wouldn't be too bad because the plugin will check again in another hour.

The `error` for  deserialization above is fine though.

Reply:
I'm not sure, this is still an error in the sense that the plugin has failed to perform the task it's intended to perform. I think an error message is appropriate, it's just not a fatal error. Happy to change though if others agree :+1:

Reply:
Yeah, that makes sense too!